keel-core/src/main/kotlin/com/netflix/spinnaker/keel/persistence/CombinedRepository.kt

```diff
@@ -50,7 +52,8 @@ class CombinedRepository(
   val artifactRepository: ArtifactRepository,
   val resourceRepository: ResourceRepository,
   override val clock: Clock,
-  override val publisher: ApplicationEventPublisher
+  override val publisher: ApplicationEventPublisher,
+  override val actuationPauser: ActuationPauser
```

Comment:
This is an example of where I see `KeelRepository` and `CombinedRepository` leading us down a path that I don't think is the right one, by creating a "catch-all" bucket where to place service-level functions, which IMHO would be better placed in smaller beans that are focused on specific sub-areas of the domain model (like `ActuationPauser` in this case). Increasingly, I fear we'll start adding dependencies to other service beans here to achieve what we need, and I don't think a "repository" is the right place to be adding that kind of logic. 🤷‍♂️

Reply:
P.S. If you feel this is not the right place to discuss this, please feel free to add your thoughts in https://github.com/spinnaker/keel/issues/927.